src/NBomber/Domain/Step.fs

```diff
@@ -14,23 +14,22 @@ open NBomber.Contracts
 open NBomber.Domain
 
 let setStepContext (correlationId: string)
-                   (feed: IFeed<obj>)
+                   (feedData: Dict<string,obj>)
                    (actorIndex: int)
                    (cancelToken: CancellationToken)
                    (logger: ILogger)
                    (step: Step) =
 
     let getConnection (pool: ConnectionPool<obj>) =
         match pool.ConnectionsCount with
-        | Some v -> let connectionIndex = actorIndex % pool.ConnectionsCount.Value
-                    pool.AliveConnections.[connectionIndex]
+        | Some v -> pool.AliveConnections.[actorIndex % v]
         | None   -> pool.AliveConnections.[actorIndex]
 
     let connection = getConnection(step.ConnectionPool)
     let context = { CorrelationId = correlationId
                     CancellationToken = cancelToken
                     Connection = connection
-                    Data = feed.GetNext()
+                    Data = feedData

```

Comment:
regarding problem of shared context.Data property
what do you think to introduce a new one property called -

context.FeedData: IReadOnlyDict // this one will be read only and used for feed
context.Data: obj   // this will be used to pass any user data

Reply:
it seems for me it's more explicit and API will be more strict.

Reply:
+ we prevent any clash btw feed and user data